sdk/python/kfp/cli/compile_.py

```diff
@@ -133,12 +134,19 @@ def parse_parameters(parameters: Optional[str]) -> Dict:
     is_flag=True,
     default=False,
     help='Whether to disable type checking.')
+@click.option(
+    '--enable-caching/--disable-caching',
+    type=bool,
```

Comment:
Make sure to use `is_flag` functionality here, to maintain precedence (just like `type_check` for instance).